File: Pdf417/src/android/java/com/phonegap/plugins/pdf417/USDLMapping.java
package com.phonegap.plugins.pdf417;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.microblink.entities.recognizers.blinkbarcode.usdl.USDLKeys;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by igrce on 14/02/2018.
 */

public final class USDLMapping {

    private Map<USDLKeys, String> mKeyMappings;

    private static volatile USDLMapping mInstance;

    private USDLMapping() {
        mKeyMappings = new HashMap<USDLKeys, String>();

        mKeyMappings.put(USDLKeys.DocumentType, "Document Type");
        mKeyMappings.put(USDLKeys.StandardVersionNumber, "Standard Version Number");
        mKeyMappings.put(USDLKeys.CustomerFamilyName, "Customer Family Name");
        mKeyMappings.put(USDLKeys.CustomerFirstName, "Customer First Name");
        mKeyMappings.put(USDLKeys.CustomerFullName, "Customer Name");
        mKeyMappings.put(USDLKeys.DateOfBirth, "Date of Birth");
        mKeyMappings.put(USDLKeys.Sex, "Sex");
        mKeyMappings.put(USDLKeys.EyeColor, "Eye Color");
        mKeyMappings.put(USDLKeys.AddressStreet, "Address - Street 1");
        mKeyMappings.put(USDLKeys.AddressCity, "Address - City");
        mKeyMappings.put(USDLKeys.AddressJurisdictionCode, "Address - Jurisdiction Code");
        mKeyMappings.put(USDLKeys.AddressPostalCode, "Address - Postal Code");
        mKeyMappings.put(USDLKeys.FullAddress, "Full Address");
        mKeyMappings.put(USDLKeys.Height, "Height");
        mKeyMappings.put(USDLKeys.HeightIn, "Height in");
        mKeyMappings.put(USDLKeys.HeightCm, "Height cm");
        mKeyMappings.put(USDLKeys.CustomerMiddleName, "Customer Middle Name");
        mKeyMappings.put(USDLKeys.HairColor, "Hair color");
        mKeyMappings.put(USDLKeys.NameSuffix, "Name Suffix");
        mKeyMappings.put(USDLKeys.AKAFullName, "Alias / AKA Name");
        mKeyMappings.put(USDLKeys.AKAFamilyName, "Alias / AKA Family Name");
        mKeyMappings.put(USDLKeys.AKAGivenName, "Alias / AKA Given Name");
        mKeyMappings.put(USDLKeys.AKASuffixName, "Alias / AKA Suffix Name");
        mKeyMappings.put(USDLKeys.WeightRange, "Weight Range");
        mKeyMappings.put(USDLKeys.WeightPounds, "Weight (pounds)");
        mKeyMappings.put(USDLKeys.WeightKilograms, "Weight (kilograms)");
        mKeyMappings.put(USDLKeys.CustomerIdNumber, "Customer ID Number");
        mKeyMappings.put(USDLKeys.FamilyNameTruncation, "Family name truncation");
        mKeyMappings.put(USDLKeys.FirstNameTruncation, "First name truncation");
        mKeyMappings.put(USDLKeys.MiddleNameTruncation, "Middle name truncation");
        mKeyMappings.put(USDLKeys.PlaceOfBirth, "Place of birth");
        mKeyMappings.put(USDLKeys.AddressStreet2, "Address - Street 2");
        mKeyMappings.put(USDLKeys.RaceEthnicity, "Race / ethnicity");
        mKeyMappings.put(USDLKeys.NamePrefix, "Name Prefix");
        mKeyMappings.put(USDLKeys.CountryIdentification, "Country Identification");
        mKeyMappings.put(USDLKeys.ResidenceStreetAddress, "Driver Residence Street Address");
        mKeyMappings.put(USDLKeys.ResidenceStreetAddress2, "Driver Residence Street Address 2");
        mKeyMappings.put(USDLKeys.ResidenceCity, "Driver Residence City");
        mKeyMappings.put(USDLKeys.ResidenceJurisdictionCode, "Driver Residence Jurisdiction Code");
        mKeyMappings.put(USDLKeys.ResidencePostalCode, "Driver Residence Postal Code");
        mKeyMappings.put(USDLKeys.ResidenceFullAddress, "Driver Residence Full Address");
        mKeyMappings.put(USDLKeys.Under18, "Under 18 Until");
        mKeyMappings.put(USDLKeys.Under19, "Under 19 Until");
        mKeyMappings.put(USDLKeys.Under21, "Under 21 Until");
        mKeyMappings.put(USDLKeys.SocialSecurityNumber, "Social Security Number");
        mKeyMappings.put(USDLKeys.AKASocialSecurityNumber, "Alias / AKA Social Security Number");
        mKeyMappings.put(USDLKeys.AKAMiddleName, "Alias / AKA Middle Name");
        mKeyMappings.put(USDLKeys.AKAPrefixName, "Alias / AKA Prefix Name");
        mKeyMappings.put(USDLKeys.OrganDonor, "Organ Donor Indicator");
        mKeyMappings.put(USDLKeys.Veteran, "Veteran Indicator");
        mKeyMappings.put(USDLKeys.AKADateOfBirth, "Alias / AKA Date of Birth");
        mKeyMappings.put(USDLKeys.IssuerIdentificationNumber, "Issuer Identification Number");
        mKeyMappings.put(USDLKeys.DocumentExpirationDate, "Document Expiration Date");
        mKeyMappings.put(USDLKeys.JurisdictionVersionNumber, "Jurisdiction Version Number");
        mKeyMappings.put(USDLKeys.JurisdictionVehicleClass, "Jurisdiction-specific vehicle class");
        mKeyMappings.put(USDLKeys.JurisdictionRestrictionCodes, "Jurisdiction-specific restriction codes");
        mKeyMappings.put(USDLKeys.JurisdictionEndorsementCodes, "Jurisdiction-specific endorsement codes");
        mKeyMappings.put(USDLKeys.DocumentIssueDate, "Document Issue Date");
        mKeyMappings.put(USDLKeys.FederalCommercialVehicleCodes, "Federal Commercial Vehicle Codes");
        mKeyMappings.put(USDLKeys.IssuingJurisdiction, "Issuing jurisdiction");
        mKeyMappings.put(USDLKeys.StandardVehicleClassification, "Standard vehicle classification");
        mKeyMappings.put(USDLKeys.IssuingJurisdictionName, "Issuing jurisdiction name");
        mKeyMappings.put(USDLKeys.StandardEndorsementCode, "Standard endorsement code");
        mKeyMappings.put(USDLKeys.StandardRestrictionCode, "Standard restriction code");
        mKeyMappings.put(USDLKeys.JurisdictionVehicleClassificationDescription, "Jurisdiction-specific vehicle classification description");
        mKeyMappings.put(USDLKeys.JurisdictionEndorsmentCodeDescription, "Jurisdiction-specific endorsment code description");
        mKeyMappings.put(USDLKeys.JurisdictionRestrictionCodeDescription, "Jurisdiction-spacific restriction code description");
        mKeyMappings.put(USDLKeys.InventoryControlNumber, "Inventory control number");
        mKeyMappings.put(USDLKeys.CardRevisionDate, "Card Revision Date");
        mKeyMappings.put(USDLKeys.DocumentDiscriminator, "Document Discriminator");
        mKeyMappings.put(USDLKeys.LimitedDurationDocument, "Limited Duration Document Indicator");
        mKeyMappings.put(USDLKeys.AuditInformation, "Audit information");
        mKeyMappings.put(USDLKeys.ComplianceType, "Compliance Type");
        mKeyMappings.put(USDLKeys.IssueTimestamp, "Issue Timestamp");
        mKeyMappings.put(USDLKeys.PermitExpirationDate, "Driver Permit Expiration Date");
        mKeyMappings.put(USDLKeys.PermitIdentifier, "Permit Identifier");
        mKeyMappings.put(USDLKeys.PermitIssueDate, "Driver Permit Issue Date");
        mKeyMappings.put(USDLKeys.NumberOfDuplicates, "Number of Duplicates");
        mKeyMappings.put(USDLKeys.HAZMATExpirationDate, "HAZMAT Endorsement Expiration Date");
        mKeyMappings.put(USDLKeys.MedicalIndicator, "Medical Indicator/Codes");
        mKeyMappings.put(USDLKeys.NonResident, "Non-Resident Indicator");
        mKeyMappings.put(USDLKeys.UniqueCustomerId, "Unique Customer Identifier");
        mKeyMappings.put(USDLKeys.DataDiscriminator, "Data discriminator");
        mKeyMappings.put(USDLKeys.DocumentExpirationMonth, "Document Expiration Month");
        mKeyMappings.put(USDLKeys.DocumentNonexpiring, "Document Nonexpiring");
        mKeyMappings.put(USDLKeys.SecurityVersion, "Security Version");

    }

    public static USDLMapping getInstance() {
        if (mInstance == null) {
            synchronized (USDLMapping.class) {
                if (mInstance == null) {
                    mInstance = new USDLMapping();
                }
            }
        }
        return mInstance;
    }

    @Nullable
    public String mapUSDLKey(@NonNull USDLKeys key) {
        return mKeyMappings.get(key);
    }
}

File: Pdf417/src/ios/sources/USDLMapping.m
//
//  USDLMapping.m
//  Pdf417Demo
//
//  Created by Jure Cular on 06/03/2018.
//

#import "USDLMapping.h"
#import <MicroBlink/MicroBlink.h>

@implementation USDLMapping

- (instancetype)init {
    self = [super init];

    if (self) {
        NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] init];

        dictionary[@(DocumentType)] = @"Document Type";
        dictionary[@(StandardVersionNumber)] = @"Standard Version Number";
        dictionary[@(CustomerFamilyName)] = @"Customer Family Name";
        dictionary[@(CustomerFirstName)] = @"Customer First Name";
        dictionary[@(CustomerFullName)] = @"Customer Name";
        dictionary[@(DateOfBirth)] = @"Date of Birth";
        dictionary[@(Sex)] = @"Sex";
        dictionary[@(EyeColor)] = @"Eye Color";
        dictionary[@(AddressStreet)] = @"Address - Street 1";
        dictionary[@(AddressCity)] = @"Address - City";
        dictionary[@(AddressJurisdictionCode)] = @"Address - Jurisdiction Code";
        dictionary[@(AddressPostalCode)] = @"Address - Postal Code";
        dictionary[@(FullAddress)] = @"Full Address";
        dictionary[@(Height)] = @"Height";
        dictionary[@(HeightIn)] = @"Height in";
        dictionary[@(HeightCm)] = @"Height cm";
        dictionary[@(CustomerMiddleName)] = @"Customer Middle Name";
        dictionary[@(HairColor)] = @"Hair color";
        dictionary[@(NameSuffix)] = @"Name Suffix";
        dictionary[@(AKAFullName)] = @"Alias / AKA Name";
        dictionary[@(AKAFamilyName)] = @"Alias / AKA Family Name";
        dictionary[@(AKAGivenName)] = @"Alias / AKA Given Name";
        dictionary[@(AKASuffixName)] = @"Alias / AKA Suffix Name";
        dictionary[@(WeightRange)] = @"Weight Range";
        dictionary[@(WeightPounds)] = @"Weight (pounds)";
        dictionary[@(WeightKilograms)] = @"Weight (kilograms)";
        dictionary[@(CustomerIdNumber)] = @"Customer ID Number";
        dictionary[@(FamilyNameTruncation)] = @"Family name truncation";
        dictionary[@(FirstNameTruncation)] = @"First name truncation";
        dictionary[@(MiddleNameTruncation)] = @"Middle name truncation";
        dictionary[@(PlaceOfBirth)] = @"Place of birth";
        dictionary[@(AddressStreet2)] = @"Address - Street 2";
        dictionary[@(RaceEthnicity)] = @"Race / ethnicity";
        dictionary[@(NamePrefix)] = @"Name Prefix";
        dictionary[@(CountryIdentification)] = @"Country Identification";
        dictionary[@(ResidenceStreetAddress)] = @"Driver Residence Street Address";
        dictionary[@(ResidenceStreetAddress2)] = @"Driver Residence Street Address 2";
        dictionary[@(ResidenceCity)] = @"Driver Residence City";
        dictionary[@(ResidenceJurisdictionCode)] = @"Driver Residence Jurisdiction Code";
        dictionary[@(ResidencePostalCode)] = @"Driver Residence Postal Code";
        dictionary[@(ResidenceFullAddress)] = @"Driver Residence Full Address";
        dictionary[@(Under18)] = @"Under 18 Until";
        dictionary[@(Under19)] = @"Under 19 Until";
        dictionary[@(Under21)] = @"Under 21 Until";
        dictionary[@(SocialSecurityNumber)] = @"Social Security Number";
        dictionary[@(AKASocialSecurityNumber)] = @"Alias / AKA Social Security Number";
        dictionary[@(AKAMiddleName)] = @"Alias / AKA Middle Name";
        dictionary[@(AKAPrefixName)] = @"Alias / AKA Prefix Name";
        dictionary[@(OrganDonor)] = @"Organ Donor Indicator";
        dictionary[@(Veteran)] = @"Veteran Indicator";
        dictionary[@(AKADateOfBirth)] = @"Alias / AKA Date of Birth";
        dictionary[@(IssuerIdentificationNumber)] = @"Issuer Identification Number";
        dictionary[@(DocumentExpirationDate)] = @"Document Expiration Date";
        dictionary[@(JurisdictionVersionNumber)] = @"Jurisdiction Version Number";
        dictionary[@(JurisdictionVehicleClass)] = @"Jurisdiction-specific vehicle class";
        dictionary[@(JurisdictionRestrictionCodes)] = @"Jurisdiction-specific restriction codes";
        dictionary[@(JurisdictionEndorsementCodes)] = @"Jurisdiction-specific endorsement codes";
        dictionary[@(DocumentIssueDate)] = @"Document Issue Date";
        dictionary[@(FederalCommercialVehicleCodes)] = @"Federal Commercial Vehicle Codes";
        dictionary[@(IssuingJurisdiction)] = @"Issuing jurisdiction";
        dictionary[@(StandardVehicleClassification)] = @"Standard vehicle classification";
        dictionary[@(IssuingJurisdictionName)] = @"Issuing jurisdiction name";
        dictionary[@(StandardEndorsementCode)] = @"Standard endorsement code";
        dictionary[@(StandardRestrictionCode)] = @"Standard restriction code";
        dictionary[@(JurisdictionVehicleClassificationDescription)] = @"Jurisdiction-specific vehicle classification description";
        dictionary[@(JurisdictionEndorsmentCodeDescription)] = @"Jurisdiction-specific endorsment code description";
        dictionary[@(JurisdictionRestrictionCodeDescription)] = @"Jurisdiction-spacific restriction code description";
        dictionary[@(InventoryControlNumber)] = @"Inventory control number";
        dictionary[@(CardRevisionDate)] = @"Card Revision Date";
        dictionary[@(DocumentDiscriminator)] = @"Document Discriminator";
        dictionary[@(LimitedDurationDocument)] = @"Limited Duration Document Indicator";
        dictionary[@(AuditInformation)] = @"Audit information";
        dictionary[@(ComplianceType)] = @"Compliance Type";
        dictionary[@(IssueTimestamp)] = @"Issue Timestamp";
        dictionary[@(PermitExpirationDate)] = @"Driver Permit Expiration Date";
        dictionary[@(PermitIdentifier)] = @"Permit Identifier";
        dictionary[@(PermitIssueDate)] = @"Driver Permit Issue Date";
        dictionary[@(NumberOfDuplicates)] = @"Number of Duplicates";
        dictionary[@(HAZMATExpirationDate)] = @"HAZMAT Endorsement Expiration Date";
        dictionary[@(MedicalIndicator)] = @"Medical Indicator/Codes";
        dictionary[@(NonResident)] = @"Non-Resident Indicator";
        dictionary[@(UniqueCustomerId)] = @"Unique Customer Identifier";
        dictionary[@(DataDiscriminator)] = @"Data discriminator";
        dictionary[@(DocumentExpirationMonth)] = @"Document Expiration Month";
        dictionary[@(DocumentNonexpiring)] = @"Document Nonexpiring";
        dictionary[@(SecurityVersion)] = @"Security Version";

        _mKeyMappings = dictionary;
    }

    return self;

}

+ (USDLMapping *)sharedInstance {
    static USDLMapping *sharedInstance = nil;
    static dispatch_once_t onceToken;

    dispatch_once(&onceToken, ^{
        sharedInstance = [[USDLMapping alloc] init];
    });

    return sharedInstance;
}

@end
